File: frontend/src/app/entitites/delivery.ts
import {Category} from "./category";
import {Person} from "./person";
import {Manager} from "./manager";
import {State} from "./state";

export class Delivery {
  constructor(
    public deliveredProduct: string,
    public quantity: number,
    public brand: string,
    public deliveryDate: string,
    public specifications: string,
    public functional: boolean,
    public category: Category,
    public person: Person,
    public manager: Manager,
    public state: State,
    public deliveryId?: number,
  ) {
  }
}

File: backend/elektro_server/src/main/java/sk/ewaste/server/entities/Delivery.java
package sk.ewaste.server.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long deliveryId;

    @Column(length = 80)
    private String deliveredProduct;

    private long quantity;

    @Column(length = 50)
    private String brand;

    private LocalDateTime deliveryDate;

    @Column(length = 100)
    private String specifications;

    private boolean functional;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "person_id")
    private Person person;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    private Manager manager;

    @ManyToOne
    @JoinColumn(name = "state_id")
    private State state;
}
